docs/source/contributing.rst

```diff
@@ -1,12 +1,210 @@
+
+If you don't have an overall idea of what the competition is like and what its
+goals are, you can brush up at the `RoboCup Wiki <http://wiki.robojackets.org/w/RoboCup>`_.
 
 Code
 ----
 
-TODO(Kevin): docstrings, GitHub, etc.
+To contribute to the codebase, you must know how to use Git, create Pull Requests, and document with docstrings. The following sections will give you some basic information about each of these, as well as addition resources to learn more about the topics.
+
+Git
+~~~
+
+We use Git as our version control system (if you already know git, you can skip
+ahead). A Version control system allows many people to code for the same project
+at the same time. A "cheat-sheet" of Git commands can be found `here <https://rogerdudler.github.io/git-guide/>`_.
+
+To ease new contributors into Git, I'll repeatedly use the analogy of the
+classroom test.
+
+If you simply want to learn about the workflow we use, and are less interested
+in learning the relationships between Git elements, you can skip to the "Overall
+Workflow" section.
+
+**Master**
+
+If Git is like a test, then you can think of the master branch as the final copy
+you submit for grading. This copy should have the correct answer, should contain
+the most effective or efficient solution, and should be highly neat and readable
+for the graders. You probably shouldn't do any work here, but rather should
+explore in other locations such as scrap paper. Once you have several solutions,
+you can pick the one you like the best, and more neatly copy the work on to the
+test you will hand in.
+
+.. figure:: https://git-scm.com/book/en/v2/images/basic-branching-1.png
+   :width: 80%
+   :align: center
+   :alt: Example of Master Branch
+
+   Example of Master Branch
+
+Our master branch which is named ``ros2`` can be found `here <https://github.com/RoboJackets/robocup-software>`_.
+The code in latest ``ros2`` is always neat and untouched. It can always compile. When you first clone our codebase from git to view the simulator and soccer, you are using the code in ``ros2``. It is in all respects, the master copy from which all other contributions are derived. Even if you are eventually given permission to write to master, you should never do so.
+
+**Branches**
+
+Branches are like pieces of scrap paper. You can use them to organize your work
+and solutions to the test questions. You should not have work regarding
+different problems mixed across several pieces of paper; you may get your
+progress confused. You should use one (or several) peices of paper for each
+problem you are trying to solve, but should never use one sheet for multiple
+questions.
+
+.. figure:: https://git-scm.com/book/en/v2/images/basic-branching-2.png
+   :width: 80%
+   :align: center
+   :alt: Creating new branch
+
+   Creating new branch
```

Comment:
I don't love this image choice--it only shows the new iss53 branch pointing to the same commit as the master branch. I feel like any of these images from further down are more instructive:

![Screenshot from 2022-06-17 11-18-20](https://user-images.githubusercontent.com/30275369/174327840-03541a29-b51a-4821-baa5-0bc6988f3b28.png)
![Screenshot from 2022-06-17 11-18-07](https://user-images.githubusercontent.com/30275369/174327841-7f9de4c4-1170-422b-91ee-df16454bd030.png)
![Screenshot from 2022-06-17 11-17-53](https://user-images.githubusercontent.com/30275369/174327842-a1d70ccd-743d-43de-b6b0-14797cea1df8.png)

Reply:
Also, the "C0" "C1" etc might be confusing to people who don't know what a "commit" is, can you work that into an explanation somewhere? It can be brief and you can reference the tutorial docs as somewhere members can go to learn more about git add/commit (since the start of those docs teach those basics)